.github/workflows/build_base.yaml

```diff
@@ -0,0 +1,148 @@
+# manually input version no.s here
+# will check in workflows to obtain 
+# this info from other files
+#
+
+name: Build base image
+
+on:
+  workflow_call:
+    inputs:
+      cmake_version:
+        type: string
+        required: false
+        default: 3.17.0
```

Comment:
Fine for now, but I think the defaults should be coming from the single source of truth (currently the NWXCMake repo) and not be hard-coded.

Reply:
Yes. As I discussed [here](https://github.com/NWChemEx-Project/ParallelZone/issues/109#issuecomment-1679664238), I will obtain the version info from files later. Now I'm planning on how to distribute the version info between repos (some packages are necessary only for certain repos).

Reply:
> Fine for now, but I think the defaults should be coming from the single source of truth (currently the NWXCMake repo) and not be hard-coded.

I'm not so sure: should I put the version info of all dependent packages in NWXCMake, or only put the versions of the general packages (used by multiple repos) in NWXCMake, and create a file in each repo to store the version info of its own dependent packages?